runtime/domain.c

```diff
@@ -436,6 +436,11 @@ asize_t caml_norm_minor_heap_size (intnat wsize)
 
   Those [young_{start,end}] variables are never accessed by another
   domain, so they need no synchronization.
+
+- Domains must decommit their minor heap segment before leaving the
```

Comment:
Shift the tone from prescriptive to descriptive.

```
Domains decommit their minor heap segment before leaving the set of STW participants. Otherwise, their minor heap arenas could get desynchronized with future changes in the reserved minor heap area. This is done by calling [free_minor_heap].
``

Meta: In the ICFP 20 paper, we use the term minor heap _arenas_ for the portion of the minor heap that is specific to a domain. This is to contrast the term with _minor heap_ which represents the union of all minor heap arenas. The runtime currently does not use this term. I'd prefer that we use arenas for clarity.